### DISCUSSION TITLE
CP-Unloader - Drescher-Priorität

### DISCUSSION THREAD

**Discussion by rohne83** [Courseplay/Courseplay_FS22#3071] [General]
Mal noch was bezüglich der CP-Abfahrer. Wenn man mehrere Abfahrer hat, und nur ein Drescher abgetankt werden muss, dafür gibts ja wohl diesen Algorithmus der aus Entfernung und Füllstand den zu rufenden Abfahrer ermittelt. Aber wie sieht das andersrum aus, also nur ein Abfahrer aber mehrere Drescher gleichzeitig die abgetankt werden müssen - gibts da auch einen Algorithmus der bestimmt in welcher Reihenfolge die Drescher anzufahren sind, oder ist das eher zufällig je nachdem wessen Meldung zuerst reinkommt?
Ich beobachte nämlich öfter mal, dass der Abfahrer dann doch erst einen weit entfernten Drescher anfährt um diesen bei 80% oder knapp darüber zu erwischen, und ein anderer Drescher der viel näher an der gegenwärtigen Abfahrerposition ist und schon einen deutlich höheren Füllstand hat (manchmal sogar schon zum Stillstand gekommen weil zu 99% voll) dabei erstmal links liegen gelassen wird. Solche Situationen gibts öfter mal beim ULW-Einsatz auf großen Feldern mit hohen Erträgen (zB Weizen, Gerste), wenn ein oder mehrere Drescher ihre 80% wieder erreichen bevor der ULW am Feldrand fertig in die AD-Abfahrer abgetankt hat. Sinnvoller wäre es in dieser Situation ja eigentlich erstmal den nahesten Drescher abzutanken, einfach auch um die zu fahrenden Wege kurz zu halten.

**Comment by Tensuko** [comment]
Der der als erstes voll wird, ruft den Abfahrer.
Da werden wir auch nichts weiter dran ändern.
Es wäre ja egal ob erst der eine anhält oder der andere, wenn man statt Füllstand auch noch die Entfernung angibt.

**Reply by rohne83** [reply]
Dann müsste nach meinem Verständnis ja aber eigentlich immer der mit dem höheren Füllstand zuerst rufen, oder irre ich mich? Wenn das so wäre, wäre es ja gut. Tut er aber wohl doch nicht immer, sonst würde oben beschriebenes Verhalten ja gar nicht auftreten.

**Reply by Tensuko** [reply]
Kann zufall sein wenn die im wechsel rufen.
Aber wie gesagt wir werden das nicht weiter aufbohren und unnötig verkomplizieren.

**Comment by nitroSkirata** [comment]
So wie ich den Ablauf verstanden hab:

Im moment rufen immer alle Drescher den Abfahrer.

Es wird der Drescher bedient, fuer den der Abfahrer als erstes losfahren muss.
Dies fuehrt dazu, dass die wahl des Dreschers zufaellig erscheint.

Beispiel:
Drescher 1: 2 min bis 80%, anfahrtzeit Abfahrer 45s ->Abfahrer muss in 1min 15 sek los
Drescher 2:  3min bis 80% Anfahrtzeit Abfahrer  2m -> Abfahrer muss in 1 min los

Abfahrer faehrt zuerst zu Drescher 2, obwohl dieser leerer ist als Drescher 1

**Comment by Tensuko** [comment]
Befreit euch von den Gedanken alles passiert gleichzeitig.
Es passiert im wechsel. Wie alles, auch der Füllstandscheck.

**Comment by rohne83** [comment]
Es geht konkret um Situation wie im Screenshot. Der Drescher direkt vor der Nase ist bereits komplett voll un zum stehen gekommen, trotzdem will CP erstmal zu dem entfernteren Drescher fahren der noch jede Menge Reserve hat.
![fsScreen_2024_03_09_17_31_40](https://github.com/Courseplay/Courseplay_FS22/assets/71030138/c1709559-f192-429b-9291-69dd81f70865)
Und das ist nur ein recht harmloses Beispiel. Das passiert auch wenn die Entfernungen schon viel größer sind, und wenn man da nicht aufpasst führt das dann schnell dazu dass der gesamte Konvoi komplett auseinanderreißt.
Ich weiß jetzt nicht wie oft das geprüft wird ob abgetankt werden muss, da ließe sich vielleicht noch was optimieren um solche Situationen zu vermeiden. Grundsätzlich wäre es aber ganz hilfreich wenn das irgendwie gequeuet würde (CP erlaubt ja eh nur maximal 5 Fahrzeuge im selben Kurs, so dass das nicht mal zu kompliziert werden müsste), so dass der Drescher mit dem höchsten filllevel auch auch mit der höchsten Priorität nen Abfahrer zugewiesen bekommt.
Oder wenn man zumindest auch ohne Debug irgendwie ersehen könnte welcher Drescher angefahren werden soll, und man die Möglichkeit hätte diese Zuweisung aufzuheben.
Problem mit der aktuellen Version ist ja auch, dass wenn der Drescher zum Stehen gekommen ist und das Schneidwerk angehoben hat,  dann mitunter nicht sauber wieder losgefahren wird sondern da gerne mal (aber auch nicht immer) bischen Frucht stehen bleibt. So dass ich es auch tunlichst vermeiden möchte dass die Drescher überhaupt anhalten müssen. Da gabs aber glaub ich schon ein Issue zu.
